core/modules/system/system.install

```diff
@@ -3148,6 +3148,24 @@ function system_update_1073() {
   }
 }
 
+/**
+ * Make sure that all config translatables are set for system core.
+ */
+function system_update_1074() {
+  $config = config('system.core');
+  $available = $config->get('_config_translatables');
+  if ($available === NULL || count($available) < 4 ) {
```

Comment:
I don't like how we use "magic" numbers here. We should be loading the default config for this module (code similar to `config_install_default_config()`), and storing the translatables in an array. Then compare that to `$available` -> add missing -> remove any duplicates -> save.

Reply:
The number (4) isn't that "magic", it's the count of array values in the array right below.

Personally, I don't think it's a problem to use "hard coded" values here, because these are all the available values at that point in this module. Loading files, of which we already know the content, seem a bit of overhead to me.

> remove duplicates...

There are no possible duplicates, we just override the complete "_config_translatables" section.

Reply:
I understand what @indigoxela is saying and it makes sense. The only downside might be if someone had a custom module that happened to also write to these configuration files. I'm not sure why someone would do that - they should be writing to its own config file - and I'm not sure if we need to take that into account.

Reply:
> The only downside might be if someone had a custom module that happened to also write to these configuration files. ... I'm not sure if we need to take that into account.

I didn't consider that... But it wouldn't be hard to do. array_merge, array_unique - should we?

Reply:
I think we should.

Reply:
> I think we should.

Done.

Reply:
The way I'm reading this is that if we ever add more translatables to `core/modules/system/config/system.core.json`, then we'd have to remember to update this update hook here again. Same with `user_update_1020()` if we ever add to `core/modules/user/config/user.mail.json`. Right? ...that's why I'm suggesting to:
- read default translatables
- read any current translatables
- compare the two and add any missing ones

Reply:
> The way I'm reading this is that if we ever add more translatables to core/modules/system/config/system.core.json, then we'd have to remember to update this update hook here again.

I don't think so. Currently translatables are missing is because we forgot to add an update hook, when they were added. There has been a proper update hook, when adding the two additional translatables for user.mail later on. We always have to add an update hook, when new translatables get added. This update hook here just fixes the oversight, when adding translatables initially (both, user.mail and system.core). So the purpose of this hook here is to fix the oversight.